Improvements to music visualization using just artwork

da-anda opened the pull request:
This adds support for the artistslideshow addon. In addition it fixes 3D depths of the viz background. Can remove the later change if desired.

I'd like to add a setting that allows to animate the slideshow in the background, wasn't sure though where to place it best and how to do this correctly. Suggestions welcome.

da-anda committed:
[Feature] support for artistslideshow add-on

da-anda commented:
I think I managed to add the setting. Just not sure about the label. Should we make it specific to artistslideshow, or do we want it more generic?
Also, would you prefer a check if the add-on is installed, and if not ask to install it like you do for other optional add-ons?

phil65 commented:
Option should get moved to Custom_1105_MusicOSDSettings.xml, and should probably apply to non-script background as well.

phil65 commented on xml/MusicVisualisation.xml:
```diff
@@ -2,25 +2,37 @@
 			<texture background="true" colordiffuse="88FFFFFF">$INFO[Player.Art(fanart)]</texture>
 			<visible>!Skin.HasSetting(hide_background_fanart)</visible>
 		</control>
+		<control type="multiimage">
```
put this control and the above one into a group and move the "shared" stuff to group level (like animations, depth)

phil65 commented on xml/MusicVisualisation.xml:
```diff
@@ -2,25 +2,37 @@
+			<loop>yes</loop>
+			<imagepath background="true">$INFO[Window(Visualisation).Property(ArtistSlideshow)]</imagepath>
+			<visible>System.HasAddon(script.artistslideshow)</visible>
+			<animation effect="zoom" start="110" end="130" center="auto" time="10000" tween="sine" easing="inout" pulse="true" condition="Skin.HasSetting(AnimateSlideshow)">Conditional</animation>
```
these 2 animations should get merged into a single one (example: https://github.com/phil65/skin.estuary/blob/master/xml/Home.xml#L37)

da-anda committed:
[visualisation] show cover/thumb as fallback when there is no fanart but user wants background artwork

da-anda commented:
updated. I tried to combine the two animations, but Kodi seems to have issues with the tweeners and easing in this case (there is no variation of the animation in any form), so I kept the two separate tags.

I added an additional commit that uses the poster/cover as fallback artwork if there is no fanart. It's looking way better than just the colored background, and if somebody doesn't want the artwork it'll also be hidden ofc.

da-anda commented:
good to go now?

da-anda commented:
bump

dagwieers commented:
[I learned yesterday](https://github.com/phil65/skin.estuary/issues/259#issuecomment-504913262) that this repository is unmaintained and you need to propose this change to the upstream https://github.com/xbmc/xbmc repository. Good luck !
